profiles/common/modules/custom/tmgmt_poetry/src/tmgmt_poetry.behat.inc

```diff
@@ -117,6 +117,76 @@ class TmgmtPoetrySubContext extends RawDrupalContext implements DrupalSubContext
   }
 
   /**
+   * Storing job reference from the Page.
+   *
+   * @Then /^I store job Reference of translation request page$/
+   */
+  public function iStoreJobReferenceOfTranslationRequestPage() {
+    $selector = ".messages";
+    $begin_message = "Status message Job has been successfully submitted for translation. Project ID is:";
+    $selector_objects = $this->getSession()->getPage()->findAll("css", $selector);
+    if (empty($selector_objects)) {
+      throw new \InvalidArgumentException("Message mismatch: the current page have not message with Job reference.");
+    }
+    foreach ($selector_objects as $selector_object) {
+      if (strpos($selector_object->getText(), $begin_message) !== FALSE) {
+        $this->jobReference = trim(substr($selector_object->getText(), strlen($begin_message)));
+        return;
+      }
+    }
+    throw new \InvalidArgumentException("Message mismatch: the current page have not message with Job reference.");
+  }
+
+  /**
+   * Test if the job reference from the page have the next version.
+   *
+   * @Then /^I test job Reference have the next version$/
+   */
+  public function iTestJobReferenceHaveTheNextVersion() {
+    // Test if a Job reference is stored.
+    if (empty($this->jobReference)) {
+      throw new \InvalidArgumentException("Job Reference mismatch: the current test have not Job Reference stored.");
+    }
+
+    $selector = ".messages";
+    $begin_message = "Status message Job has been successfully submitted for translation. Project ID is:";
+    $selector_objects = $this->getSession()->getPage()->findAll("css", $selector);
```

Comment:
Same as comment on `iStoreJobReferenceOfTranslationRequestPage()`: we could get the latest project ID from the database.